Mathlib/Topology/Category/FinTopCat.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.Topology.Category.TopCat.Basic
+
+/-!
+# Category of finite topological spaces
+
+Definition of the category of finite topological spaces with the canonical
+forgetful functors.
+
+-/
+
+
+universe u
+
+open CategoryTheory
+
+/-- A bundled finite topological space. -/
+structure FinTopCat where
+  /-- carrier of a finite topological space. -/
+  α : Type u
+  [fintype : Fintype α]
+  [topologicalSpace : TopologicalSpace α]
+
+namespace FinTopCat
+
+instance : Inhabited FinTopCat :=
+  ⟨⟨PUnit⟩⟩
+
+instance : CoeSort FinTopCat (Type u) :=
+  ⟨FinTopCat.α⟩
+
+attribute [instance] fintype topologicalSpace
+
+instance : Category FinTopCat.{u} where
+  Hom X Y := { f : X → Y // Continuous f }
```

Comment:
You probably want to use `C(X, Y)` instead. An even simpler solution would be to use `InducedCategory.category` to inherit the category instance from `TopCat`, but then you need to change the definition of `FinTopCat` so that one of the fields is an object of `TopCat`. This gives you lots of the stuff below for free, you can compare with the definitions of `CompHaus` and `Profinite` for example.